source/release-notes/v2.1-release-notes.rst

```diff
@@ -0,0 +1,155 @@
+.. _v2.1-release-notes:
+
+v2.1 Release Notes
+==================
+
+.. warning::
+
+   There are some breaking changes in 2.1. See the upgrade directions below for details.
+
+
+Highlights in 2.1:
+
+- `Dynamic Javascript in BatchConnect Forms`_
+- `Dependency updates`_
+
+Upgrading from v2.0
+-------------------
+
+Breaking Changes
+................
+
+
+context.json file locations have changed
+****************************************
+
+In versions 2.0 and below, batch connect apps wrote a `context.json` file to
+a directory like ``~/data/sys/dashboard/batch_connect/sys/<APPNAME>/context.json``.
+OnDemand uses these files to cache the choices a user makes for the next time they
+use that app.
+
+Version 2.1 now writes files like ``~/ondemand/data/sys/dashboard/batch_connect/cache/<APPNAME>.json``
+
+When you upgrade, since no files exist in a users cache directory, users will have no
+cached settings in any of their batch connect apps.
+
+
+Upgrade directions
+..................
+
+.. warning::
+
+   As always please update the *development* or *test* instances of OnDemand installed at your center first to test and verify before you modify the *production* instance.
+
+.. warning::
+
+   The OnDemand upgrade has only been tested going from 2.0.x to 2.1.x.
+
+#. Update OnDemand release RPM
+
+   .. code-block:: sh
+
+      sudo yum install -y https://yum.osc.edu/ondemand/2.0/ondemand-release-web-2.1-1.noarch.rpm
+
+#. Enable dependency repos
+
+   **CentOS/RHEL 8 only**
+
+   .. code-block:: sh
+
+      sudo dnf module reset ruby
+      sudo dnf module enable ruby:2.7
+      sudo dnf module reset nodejs
+      sudo dnf module enable nodejs:14
```

Comment:
```suggestion
      sudo dnf module reset nodejs
      sudo dnf module enable nodejs:14
```

Technically reset of Ruby to 2.7 isn't needed since if upgrade from 2.0, already using Ruby 2.7.  Maybe we say must upgrade from 2.0 first?